### PR TITLE
[FEATURE] Add interface for configurable rule

### PR DESCRIPTION
**Pull request by sabbelasichon**

### CODE REVIEW

**Comment by sabbelasichon** [comment]
@andreaswolf Btw. that´s the reason why the guard clause https://github.com/andreaswolf/fractor/blob/main/fractor-doc-generator/src/Printer/CodeSamplePrinter.php#L54 exists

**Comment by andreaswolf** [comment]
> @andreaswolf Btw. that´s the reason why the guard clause https://github.com/andreaswolf/fractor/blob/main/fractor-doc-generator/src/Printer/CodeSamplePrinter.php#L54 exists

That part is still a mystery to me, but I guess it's because I've never seen an example in action. Once I (think I) understood it, I'll add documentation where appropriate :)

File: fractor/src/Application/Contract/ConfigurableFractorRule.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+declare(strict_types=1);
+
+namespace a9f\Fractor\Application\Contract;
+
+use Symplify\RuleDocGenerator\Contract\ConfigurableRuleInterface;
+
+interface ConfigurableFractorRule extends FractorRule, ConfigurableRuleInterface
+{
+    /**
+     * @param mixed[] $configuration
```
Comment (andreaswolf) [inline fractor/src/Application/Contract/ConfigurableFractorRule.php:12]: The configuration should always be a map of string => mixed, right? Maybe we can at least have that little type info here.

**Review by andreaswolf — COMMENTED** [review]
Looks good IMO apart from the pretty generic type info :)

File: fractor/src/Application/Contract/ConfigurableFractorRule.php (hunk 1 above)
Comment (sabbelasichon) [reply fractor/src/Application/Contract/ConfigurableFractorRule.php:12]: It can also be an integer IMHO

**Pull request merged**